### DISCUSSION TITLE
0006 XLS-6d Standard for Visual Account Icons

### DISCUSSION THREAD

**Discussion by RichardAH** [XRPLF/XRPL-Standards#24] [XLS Proposals]
## The earlier version can be found by looking at the edit history.

**Comment by RichardAH** [comment]
Example: 
r-addr: r3kmLJN5D28dHuH8vZNUZpMC43pEHpaocV
x-addr: X7tFPvjMH7nDxP8nTGkeeggcUpCZj8nYJkTRDHUikbvmVMB
hashicon: 
![image](https://user-images.githubusercontent.com/19866478/65388972-3c6a2b80-dd94-11e9-9ef3-d8e64ca16d63.png)

r-addr: rGWrZyQqhTp9Xu7G5Pkayo7bXjH4k4QYpf
x-addr: XVLhHMPHU98es4dbozjVtdWzVrDjtV5fdx1mHp98tDMoQXb
hashicon:
![image](https://user-images.githubusercontent.com/19866478/65388989-691e4300-dd94-11e9-8229-0a81b163b776.png)

**Comment by movitto** [comment]
@codetsunami I'm all for visual representations of data but is the hashicon algorithm / standard documented anywhere outside the provided repo? I ask because unless it can be implemented in other languages, this would be limited to javascript based solutions.

Also googling around a bit there are a few other solutions in the same domain, such as [identicons](https://en.wikipedia.org/wiki/Identicon), just curious as to if you've evaluated the tradeoffs?

**Comment by RichardAH** [comment]
@movitto I had a look at the various other identicons on offer. None seem as visually appealing and easy to recognise as these hashicons.

I take your point about support for other languages. I’m happy to port the algorithm that generates them to a couple of other languages if you let me know what you need.

**Comment by movitto** [comment]
> @movitto I had a look at the various other identicons on offer. None seem as visually appealing and easy to recognise as these hashicons.

"None seem as visually appealing" isn't really an objective argument, to support that I'd reference color and pattern variations, algorithm differences, etc. 

> I take your point about support for other languages. I’m happy to port the algorithm that generates them to a couple of other languages if you let me know what you need.

Alot of our stack is in ruby, so would be awesome to have bindings for that. That being said, C bindings probably would be the most portable though, as once those exist, wrappers could be written in any language.

**Comment by RichardAH** [comment]
@movitto if you want to propose a different identicon as being superior please go ahead. At the end of the day everyone will have a different opinion on any graphical representation, there’s no way to come up with an “objectively best” visual.

**Comment by ihomp** [comment]
I've seen it added in Toast Wallet already, I've added it in Bithomp as well. I like the idea.

**Comment by mvadari** [comment]
@RichardAH @ihomp should this spec be moved into PR and kept in the repo or should it be closed? Note that this does not mean that the spec has been accepted/implemented/finalized.

I will move it to PR by copy-pasting the existing draft if there is no response in the next week.
